Dataverse Chat Readme en portugues (#1633)
Laboratoria/curriculum: projects/03-dataverse-chat/README.md

## Índice

> * [1. Preâmbulo](#1-preâmbulo)

Author: Substituí o README original para facilitar a visualização do diff, antes de mesclar criarei um novo arquivo README.pt.md

* [2. Resumo do projeto](#2-resumo-do-projeto)
* [3. Considerações gerais](#3-considerações-gerais)
* [4. Funcionalidades](#4-funcionalidades)
* [5. Considerações técnicas](#5-considerações-técnicas)
* [6. Critérios de aceitação mínimos do projeto](#6-critérios-de-aceitação-mínimos-do-projeto)
* [7. Objetivos de aprendizagem](#7-objetivos-de-aprendizagem)

> * [8. Dicas, dicas e leituras complementares](#8-dicas-dicas-e-leituras-complementares)

Author: ```suggestion
* [8. Dicas, guias e leituras complementares](#8-dicas-guias-e-leituras-complementares)
```

* [9. Considerações para pedir seu Project Feedback](#9-considerações-para-pedir-seu-project-feedback)

***

## 1. Preâmbulo

No nosso dia a dia, as aplicações web são como amigas inseparáveis. Desde as redes sociais até as ferramentas de trabalho, nós as utilizamos o tempo todo. Mas o que torna este momento realmente especial é que essas aplicações podem nos abrir uma porta para algo incrível: a interação com a inteligência artificial.

E se pudéssemos conversar com alguém que viveu há mais de cem anos ou até mesmo com os personagens de nossas séries e filmes favoritos, ou com entidades que normalmente não podem falar, como por exemplo planetas ou animais? A inteligência artificial pode nos ajudar a entender o passado e nos conectar com figuras históricas, assim como nos imergir em mundos fictícios.

![Visualização do aplicativo](https://github-production-user-asset-6210df.s3.amazonaws.com/123121338/271433237-2bd1477b-15ef-49d4-9fcb-226b3263c46a.png)

## 2. Resumo do projeto

> Neste projeto, você irá transformar a aplicação desenvolvida no Dataverse em uma [Single Page Application (SPA)](https://pt.wikipedia.org/wiki/Aplicação_de_página_única), mantendo as funcionalidades de visualização, filtragem, ordenação e cálculo de alguma estatística, adicionando uma nova visualização para consultar informações detalhadas de cada personagem/entidade e adicionando a possibilidade de interagir com um personagem/entidade ou todos eles através de um sistema de chat impulsionado pela [API da OpenAI](https://openai.com/product).

Reviewer: Aqui se dice que si deve crear un app que pueda interagir con `un personaje` o todos... 
pero más abajo, en `4.funcionalidades` es pedido que se tenga un chat que puede interagir con **todos** y me quedé confundida.. Es un unico chat para todos o puede ser individual tambien?

> Neste projeto, você irá transformar a aplicação desenvolvida no Dataverse em uma [Single Page Application (SPA)](https://pt.wikipedia.org/wiki/Aplicação_de_página_única), mantendo as funcionalidades de visualização, filtragem, ordenação e cálculo de alguma estatística, adicionando uma nova visualização para consultar informações detalhadas de cada personagem/entidade e adicionando a possibilidade de interagir com um personagem/entidade ou todos eles através de um sistema de chat impulsionado pela [API da OpenAI](https://openai.com/product).

Author: Los dos, puede hacer el chat con uno individual, o tambien con todos en modo "Panel"

> Neste projeto, você irá transformar a aplicação desenvolvida no Dataverse em uma [Single Page Application (SPA)](https://pt.wikipedia.org/wiki/Aplicação_de_página_única), mantendo as funcionalidades de visualização, filtragem, ordenação e cálculo de alguma estatística, adicionando uma nova visualização para consultar informações detalhadas de cada personagem/entidade e adicionando a possibilidade de interagir com um personagem/entidade ou todos eles através de um sistema de chat impulsionado pela [API da OpenAI](https://openai.com/product).

Author: Puede hacer chat uno-uno idividual y a todos en modo Panel

```text
Usuária: "Olá, Ada. Conta-me mais sobre as tuas contribuições para a informática."

Ada Lovelace (AI da OpenAI): "Olá! Claro, ficarei encantada em falar sobre isso. As minhas contribuições mais destacadas estão relacionadas ao trabalho com Charles Babbage no design da Máquina Analítica. Além disso, desenvolvi o primeiro algoritmo destinado a ser processado por uma máquina, o que me torna a primeira programadora da história. Há algo mais em particular que gostarias de saber?"

Usuária: "Isso é fascinante! Que desafios enfrentaste naquela época?"

Ada Lovelace (AI da OpenAI): "Naquela época, enfrentei vários desafios, um dos principais foi a limitação de recursos e a falta de reconhecimento do potencial da Máquina Analítica. Também tive que superar as restrições sociais da época, já que as mulheres tinham acesso limitado à educação em matemática e ciências. No entanto, minha paixão pela matemática e a visão do que essas máquinas poderiam realizar me impulsionaram a seguir em frente."

```

> * A aplicação deve permitir que o usuário interaja de forma simultânea com **todos** os personagens/entidades através de um chat:

Reviewer: Aqui si dice chat para todos

> Aqui se dice que si deve crear un app que pueda interagir con `un personaje` o todos... 
pero más abajo, en `4.funcionalidades` es pedido que se tenga un chat que puede interagir con **todos** y me quedé confundida.. Es un unico chat para todos o puede ser individual tambien?

Author: SI - es que puede hacer chat uno-uno o con todos.

Author: SI y puede hacerlo individual tambien
https://laboratoria-dataverse-talks.netlify.app/

### `src/components`

Esta pasta conterá arquivos JavaScript que representam [componentes](https://lenguajejs.com/vuejs/componentes/que-es-un-componente/) reutilizáveis da sua aplicação. Esses componentes podem ser elementos da interface do usuário usados em várias partes da sua aplicação. Organizar seus componentes dessa maneira ajuda a manter um código mais modular e reutilizável.

### `src/data`

Nesta pasta estão os dados com os quais você vai trabalhar (dados de exemplo ou dados que serão gerados com a ajuda da inteligência artificial).

### `src/lib`

> Nesta pasta, você criará módulos que contêm funcionalidades não relacionadas ao DOM. Aqui, você poderá incluir o arquivo `dataFunctions.js`, que contém as funções que permitem filtrar, ordenar e fazer cálculos agregados. Você também poderá criar arquivos com as funções que interajam com a API de inteligência artificial ou utilidades destinadas a armazenar dados no armazenamento local.

Reviewer: ```suggestion
Nesta pasta, você criará módulos que contêm funcionalidades não relacionadas ao DOM. Aqui, você poderá incluir o arquivo `dataFunctions.js`, que contém as funções que permitem filtrar, ordenar e fazer cálculos agregados. Você também poderá criar arquivos com as funções que interajam com a API de inteligência artificial ou utilidades destinadas a armazenar dados no local storage.
```

### `src/index.js`

O ponto de partida para a sua SPA será o arquivo `src/index.js`. Aqui, você encontrará algumas instruções e comentários para orientá-lo no início do desenvolvimento da sua aplicação.

### `src/router.js`

Neste arquivo, você irá gerenciar o roteamento da sua aplicação. É um arquivo que você terá que criar por conta própria.

### `src/style.css`

É um arquivo que você terá que criar por conta própria para adicionar os estilos necessários para a sua aplicação.

### `src/test`

Esta pasta contém os testes das suas funções e componentes. Por exemplo, os testes das `dataFunctions` devem estar aqui.

## 6. Critérios de aceitação mínimos do projeto

Além das funcionalidades que definimos na seção de [Funcionalidades](#4-funcionalidades), o seu projeto deve atender aos seguintes requisitos:

### Definição do produto

Documente brevemente o seu trabalho no arquivo `README.md` do seu repositório, contando como foi o seu processo de design e como você acredita que o produto resolve o problema (ou problemas) da sua usuária.

### Histórias de usuária

> Depois de entender as necessidades das suas usuárias, escreva as [Histórias de Usuária](https://es.wikipedia.org/wiki/Historias_de_usuario) que representam tudo o que a usuária precisa fazer/ver. As **Histórias de Usuária** devem ser o resultado do seu processo de pesquisa ou _research_ das suas usuárias.

Reviewer: ```suggestion
Depois de entender as necessidades das suas usuárias, escreva as [Histórias de Usuária](https://pt.wikipedia.org/wiki/História_de_usuário) que representam tudo o que a usuária precisa fazer/ver. As **Histórias de Usuária** devem ser o resultado do seu processo de pesquisa ou _research_ das suas usuárias.
```

Certifique-se de incluir a definição de pronto (_definition of done_) e os Critérios de Aceitação para cada uma.

> Use as suas histórias de usuária para planejar os seus sprints dividindo cada história em tarefas.

Reviewer: ```suggestion
Use as suas histórias de usuária para planejar as suas sprints dividindo cada história em tarefas.
```

### Design da Interface de Usuária

#### Protótipo de baixa fidelidade

Durante o seu trabalho, você deverá ter feito e iterado esboços (_sketches_) da sua solução usando papel e lápis. Recomendamos que tire fotos de todas as iterações que fizer, as suba para o seu repositório e as mencione no seu `README.md`.

Lembre-se de pedir feedback do seu protótipo para as suas colegas e/ou orientadores.

#### Protótipo de alta fidelidade

O próximo passo é projetar a sua Interface de Usuária (UI - _User Interface_). Para isso, você deve aprender a utilizar alguma ferramenta de design visual. Nós recomendamos o [Figma](https://www.figma.com/), que é uma ferramenta que funciona no navegador e, além disso, você pode criar uma conta gratuita. No entanto, você é livre para usar outros editores gráficos como Illustrator, Photoshop, etc.

O design deve representar o _ideal_ da sua solução. Digamos que é o que você gostaria de implementar se tivesse tempo ilimitado para trabalhar. Além disso, o seu design deve seguir os fundamentos de _visual design_.

> Lembre-se de pedir feedback do seu protótipo para as suas colegas e/ou orientadores.

Reviewer: ```suggestion
Lembre-se de pedir feedback do seu protótipo para as suas colegas e/ou coaches.
```

## 7. Objetivos de aprendizagem

> ℹ️ Esta seção será automaticamente gerada no idioma apropriado, com base nos objetivos de aprendizagem declarados em [`project.yml`](./project.yml), ao criar o repositório do projeto para uma turma específica usando [`./scripts/create-cohort-project.js`](../../scripts#create-cohort-project-coaches).

> Aqui você pode conferir uma [lista de todos os objetivos de aprendizagem](../../learning-objectives/data.yml) que contempla nossa grade curricular.

## 8. Dicas, dicas e leituras complementares

### Primeiros passos

Junte-se ao canal Slack [#project-dataverse-chat](https://claseslaboratoria.slack.com/archives/C03MV35EP5M) para conversar e pedir ajuda sobre o projeto.

Sugerimos algumas tarefas para iniciar o projeto. Não é necessário abordá-las de maneira sequencial e você pode realizá-las em paralelo enquanto avança no planejamento das Histórias do Usuário:

#### Design de Wireframes ou Protótipo de Baixa Fidelidade

> Comece projetando wireframes ou protótipos de baixa fidelidade para capturar os requisitos funcionais do seu projeto. É recomendável começar com uma abordagem ["mobile first"](https://mediaclick.es/blog/diseno-web-responsive-design-y-la-importancia-del-mobile-first/) para garantir usabilidade em dispositivos móveis e, em seguida, estender a atenção para telas maiores, como as de desktop.

Author: Para mi, me carga

> Comece projetando wireframes ou protótipos de baixa fidelidade para capturar os requisitos funcionais do seu projeto. É recomendável começar com uma abordagem ["mobile first"](https://mediaclick.es/blog/diseno-web-responsive-design-y-la-importancia-del-mobile-first/) para garantir usabilidade em dispositivos móveis e, em seguida, estender a atenção para telas maiores, como as de desktop.

Author: funciona para mi

#### Quando estiver pronta para codificar, sugerimos começar assim

1. Um dos membros da equipe deve fazer um :fork_and_knife: [fork](https://help.github.com/articles/fork-a-repo/) do repositório da sua turma; os _coaches_ vão compartilhar um _link_ para um repositório e conceder acesso de leitura a esse repositório. O outro membro da equipe deve fazer um fork **do repositório da colega** e [configurar](https://gist.github.com/BCasal/026e4c7f5c71418485c1) um `remote` para o mesmo.
2. :arrow_down: [Clone](https://help.github.com/articles/cloning-a-repository/) o seu _fork_ para o seu computador (cópia local).
3. 📦 Instale as dependências do projeto com o comando `npm install`. Isso pressupõe que você tenha instalado o [Node.js](https://nodejs.org/) (que inclui [npm](https://docs.npmjs.com/)).
4. Se tudo correr bem, você deverá ser capaz de executar os :traffic_light: testes unitários (unit tests) com o comando `npm test`.
5. Para visualizar a interface do seu programa no navegador, use o comando `npm start` para iniciar o servidor web e acesse `http://localhost:5000` no seu navegador.
6. Hora de codificar! :rocket:

### Conteúdo de referência

#### Desenvolvimento Front-end

> * [Tópicos no currículo do site da Laboratoria](https://curriculum.laboratoria.la/es/web-dev/topics) testes, arrays, objetos, funções, DOM no Browser Javascript.

Reviewer: ```suggestion
* [Tópicos no currículo do site da Laboratoria](https://curriculum.laboratoria.la/pt/web-dev/topics) testes, arrays, objetos, funções, DOM no Browser Javascript.
```

> * [Promises](https://es.javascript.info/promise-basics)

Reviewer: Link em pt?

Author: hay algo en ingles, mejor?

Author: no aun. voy a cambiarlo por ingles

> * [Promises](https://es.javascript.info/promise-basics)

Author: ```suggestion
* [Promises](https://javascript.info/promise-basics)
```

> * [Promises](https://es.javascript.info/promise-basics)

Author: ```suggestion
* [Promises](https://javascript.info/promise-basics)
```

* [Fazendo requisições com Fetch](https://developer.mozilla.org/es/docs/Web/API/Fetch_API/Utilizando_Fetch)

> * [Armazenamento com LocalStorage](https://es.javascript.info/localstorage)

Reviewer: link em pt?

Author: voy a poner el ingles

Author: voy a cambiarlo a ingles

> * [Armazenamento com LocalStorage](https://es.javascript.info/localstorage)

Author: ```suggestion
* [Armazenamento com LocalStorage](https://javascript.info/localstorage)
```

> * [Armazenamento com LocalStorage](https://es.javascript.info/localstorage)

Author: ```suggestion
* [Armazenamento com LocalStorage](https://javascript.info/localstorage)
```

> * [Criando um roteador SPA com Javascript puro](https://github.com/Laboratoria/guide-router/blob/guide-v1/README.md)

Reviewer: link em pt?

Author: Aun no, pero ahora voy a crear un issue para traducirlo

Author: aun no pero acabo de crear un issue #1651

Author: Si te interesa hacer un PR 😉 https://github.com/Laboratoria/guide-router/issues/3

> * [Delegação de eventos](https://es.javascript.info/event-delegation)

Reviewer: link em pt?

> * [Delegação de eventos](https://es.javascript.info/event-delegation)

Author: ```suggestion
* [Delegação de eventos](https://javascript.info/event-delegation)
```

* [Aprenda sobre Prompting](https://learnprompting.org/pt/docs/category/-basic-applications)
* [Roadmap de Engenharia de Prompting](https://roadmap.sh/prompt-engineering)
* [OpenAI Playground](https://beta.openai.com/playground)
* [Fazendo requisições](https://platform.openai.com/docs/api-reference/making-requests)
* [Conversando com IA](https://platform.openai.com/docs/api-reference/chat)
* [Priming para Chatbots](https://learnprompting.org/es/docs/basics/priming_prompt)
* [Construa sua aplicação com IA](https://platform.openai.com/docs/quickstart/build-your-application)

#### Ferramentas

* [Jest](https://jestjs.io/)
* [Vercel](https://vercel.com/)
* [Netlify](https://www.netlify.com/)

## 9. Considerações para solicitar seu Feedback do Projeto

Antes de agendar seu Feedback do Projeto com um coach, certifique-se de que seu projeto:

* [ ] Cumpre todas as testes unitários ao executar `npm run test` e possui uma cobertura de 70% de _statements_ (_declarações_), _functions_ (_funções_), _lines_ (_linhas_) e _branches_ (_ramificações_).
* [ ] Está livre de _erros_ do `eslint` ao executar `npm run pretest`.
* [ ] Está enviado para o GitHub e implantado no Netlify ou Vercel.

> * Possui um `README.md` com

Author: ```suggestion
* Tem um `README.md` com o seguinte:
  - [ ] _Definição do produto_ clara e informativa
  - [ ] Histórias de usuário
  - [ ] Um _sketch_ da solução (protótipo de baixa fidelidade) e
    _Design de Interface do Usuário_ (protótipo de alta fidelidade)
  - [ ] Lista de problemas detectados por testes de usabilidade no `README.md`
* Tem uma UI que atende às funcionalidades:
  - [ ] Permite visualizar todos os dados na tela principal com opções para
    filtrar e ordenar.
  - [ ] Permite inserir uma chave de API por meio de um formulário.
  - [ ] Permite visualizar uma página com mais detalhes de cada entidade. Nesta página,
    é possível iniciar o chat.
  - [ ] Permite conversar com o personagem ou entidade dos dados por meio
    de um chat.
  - [ ] Permite conversar de forma grupal com todas as entidades dos dados.
  - [ ] É responsivo e foi realizado usando a metodologia mobile first.

Lembre-se de fazer uma autoavaliação de _objetivos de aprendizado_ e
_habilidades pessoais_ a partir do seu painel de estudante.

Se você não concluiu tudo o que foi mencionado acima, não consideramos que você está pronta para
sua sessão de Project Feedback.

```

> * Possui um `README.md` com

Reviewer: creo q está cortado

Author: el ChatGPT estaba cansado
